### PR TITLE
Implement Authorization Rules for Post and Comment Deletion in Blog App

### PR DESCRIPTION
**Pull request by axcell-0**
Overview
This pull request addresses the project requirements related to authorization rules for post and comment deletion in the Blog app. The key changes include the installation of CanCanCan, addition of a role column to the users table, and implementation of authorization logic for post and comment deletion based on user roles.

Changes Made
Installed CanCanCan:

CanCanCan has been successfully installed in the project to handle authorization.
Added Role Column to Users Table:

A migration has been created and applied to add a role column to the users table.
Deletion Authorization:

Implemented authorization rules using CanCanCan.
Users can now delete a post if it belongs to them or if they have an admin role ("admin" value in the role column).
Added a "Delete" button.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
```
Comment (V-Blaze) [inline app/models/ability.rb:3]: - You have done great so far :+1: :+1: But there are some requirements you need to meet before moving to the next milestone. :+1: :100: 
`Install CanCanCan in your project.`

- [ ] According to the project requirement, you were meant to install `cancancan` in order to be able to use the authorization rules. But looking at your `Gem File`, the gem is not present. I'll suggest you install the appropriate gem and make sure your project works as expected. :+1: :100:

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,313 @@
+# frozen_string_literal: true
```
Comment (V-Blaze) [inline config/initializers/devise.rb:1]: - [ ] While trying to create the DB and start the project, I got this error as seen in the screenshot below. The error points at an `uninitialized constant Devise`. I'll suggest you fix this as it prevents your project from running. :+1: :100: :+1:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @sam2if, @axcell-0  

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Added role column to user table✔
- Correct git flow👋 
-  Meaningful commit messages✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi team,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**